src/main/java/com/minecolonies/coremod/client/gui/WindowResearchTree.java

```diff
@@ -137,299 +199,775 @@ public void onButtonClicked(@NotNull final Button button)
     /**
      * Draw the tree of research.
      *
-     * @param height           the start y offset.
-     * @param depth            the current depth.
-     * @param view             the view to append it to.
-     * @param researchList     the list of research to go through.
-     * @param tree             the local tree of the colony.
-     * @param parentResearched if possibly can be researched.
-     * @param abandoned        if abandoned child.
-     * @param parentHeight     the height of the parent.
+     * @param height                 the start y offset.
+     * @param depth                  the current depth.
+     * @param view                   the view to append it to.
+     * @param researchList           the list of research to go through.
+     * @param abandoned              if abandoned child.
      * @return the next y offset.
      */
     public int drawTree(
       final int height,
       final int depth,
       final ZoomDragView view,
       final List<String> researchList,
-      final ILocalResearchTree tree,
-      final boolean parentResearched,
-      final boolean abandoned,
-      final int parentHeight)
+      final boolean abandoned)
     {
         int nextHeight = height;
         for (int i = 0; i < researchList.size(); i++)
         {
             final String researchLabel = researchList.get(i);
-            int offsetX = (depth * (GRADIENT_WIDTH + X_SPACING));
 
             final IGlobalResearch research = IGlobalResearchTree.getInstance().getResearch(branch, researchLabel);
-            final ILocalResearch localResearch = tree.getResearch(branch, research.getId());
-            final ResearchState state = localResearch == null ? ResearchState.NOT_STARTED : localResearch.getState();
-            final IGlobalResearch parentResearch = IGlobalResearchTree.getInstance().getResearch(branch, research.getParent());
+            if (research.isHidden() && !IGlobalResearchTree.getInstance().isResearchRequirementsFulfilled(research.getResearchRequirement(), this.building.getColony()))
+            {
+                continue;
+            }
+
+            final int offsetX = (depth * (GRADIENT_WIDTH + X_SPACING));
+            final int offsetY = (nextHeight + Math.min(i, 1)) * (GRADIENT_HEIGHT + Y_SPACING) + Y_SPACING + TIMELABEL_Y_POSITION;
```

Comment:
please, try to remove `Math.min(i, 1)` from logic of this method

Reply:
Ok.  Was in original variant, but wasn't if something to keep.